### DISCUSSION TITLE
Feature Request: Listen for TCP/HTTP

### DISCUSSION THREAD

**Discussion by mickmister** [microsoft/devicescript#579] [General]
Hi, I'm trying to have my microcontroller host an http server, but I'm not seeing anything in the docs related to listening for TCP connections.

Is it possible to listen for TCP connections? If not, is it feasible to have this implemented? Interested in contributing effort for this if so. Thank you!

**Comment by pelikhan** [comment]
Yes we have TCP and TLS! https://microsoft.github.io/devicescript/developer/net#tcptls-sockets

**Comment by mmoskal** [comment]
Unfortunately while you can connect to servers you cannot listen. We are also currently limited to one socket at a time.

**Reply by mickmister** [reply]
@mmoskal I imagine this is currently not possible because of some technical limitations. Do you think it's likely the support this in the future? This would be amazing 🤩

**Reply by mickmister** [reply]
Also good to know about the "one socket at a time". Thanks for that :+1:

**Reply by mmoskal** [reply]
It's definitely possible, but don't hold your breath... I think we'll focus on better supporting the client scenarios first - it's more and more difficult to host a plain http server on local network - the ever-tightening TLS and other security requirements are working against you.
